diff --git a/PCbuild/openssl.props b/PCbuild/openssl.props
index a4186f01da..e3db93d612 100644
--- a/PCbuild/openssl.props
+++ b/PCbuild/openssl.props
@@ -10,7 +10,7 @@
     </Link>
   </ItemDefinitionGroup>
   <PropertyGroup>
-    <_DLLSuffix>-3</_DLLSuffix>
+    <_DLLSuffix>-1_1</_DLLSuffix>
     <_DLLSuffix Condition="$(Platform) == 'ARM'">$(_DLLSuffix)-arm</_DLLSuffix>
     <_DLLSuffix Condition="$(Platform) == 'ARM64'">$(_DLLSuffix)-arm64</_DLLSuffix>
     <OpenSSLDLLSuffix Condition="$(OpenSSLDLLSuffix) == ''">$(_DLLSuffix)</OpenSSLDLLSuffix>
